### PR TITLE
Use psql as a service

### PR DESCRIPTION
**Pull request by changjy44**
Refactor local postgres to psql as a service.

### CODE REVIEW

**Review by clevon-w — APPROVED** [review]
lgtm

File: src/backend/db.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
+const host = process.env.PSQL_HOST;
+const password = process.env.PSQL_PASSWORD;
+const user = process.env.PSQL_USER;
+const database = process.env.PSQL_DATABASE;
 
 const pool = new Pool({
-    user: "postgres",
-    password: "123",
-    host: "localhost",
+    user: user,
```
Comment (ish1506) [inline src/backend/db.js]: You can use this JS shorthand for `user`, `password`, `host` and `database`

```suggestion
    user,
```

File: src/backend/db.js [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+    password: password,
+    host: host,
     port: 5432,
-    database: "userservice"
+    database: database
```
Comment (ish1506) [inline src/backend/db.js]:
```suggestion
    database
```

File: src/backend/db.js [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
+const password = process.env.PSQL_PASSWORD;
+const user = process.env.PSQL_USER;
+const database = process.env.PSQL_DATABASE;
 
 const pool = new Pool({
-    user: "postgres",
-    password: "123",
-    host: "localhost",
+    user: user,
+    password: password,
```
Comment (ish1506) [inline src/backend/db.js:11]:
```suggestion
```

File: src/backend/db.js [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
+    host: host,
```
Comment (ish1506) [inline src/backend/db.js:12]:
```suggestion
```

**Review by ish1506 — COMMENTED** [review]
LGTM, I left a suggestion

**Review by clevon-w — APPROVED** [review]
tested liaoss

File: src/user-service/utils/validator.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const pool = require("../db.js");
+
+exports.checkIfNameOrEmailExists = async (username, email) => {
+    const user = await pool.query("SELECT * FROM Users WHERE username = $1 OR email = $2", [username, email]);
+    
+    if (user.rows.length > 0) {
```
Comment (ish1506) [inline src/user-service/utils/validator.js]:
```suggestion
    return user.rows.length > 0;
```
This might be cleaner

**Review by ish1506 — COMMENTED** [review]
I left a suggestion. Maybe you could standardise the tab size to be 2 for all files too?

**Review by panpannnnn — APPROVED** [review]
lgtm after basic testing

**Pull request merged**